Osiris/Hooks.cpp

```diff
@@ -327,14 +327,20 @@ static bool __stdcall fireEventClientSide(GameEvent* event) noexcept
     if (event) {
         switch (fnv::hashRuntime(event->getName())) {
         case fnv::hash("player_death"):
+            Misc::teamDamageCounter(event);
             Misc::killMessage(*event);
             SkinChanger::overrideHudIcon(*event);
             break;
         case fnv::hash("player_hurt"):
+            Misc::teamDamageCounter(event);
             Misc::playHitSound(*event);
-            Visuals::hitEffect(event);                
+            Visuals::hitEffect(event);
             Visuals::hitMarker(event);
             break;
+        case fnv::hash("round_announce_match_start"):
```

Comment:
This dispatch could be moved to `Misc::teamDamageCounter` in order to not involve `Misc::teamKills` and ` Misc::teamDamage` here.

Reply:
currently `Misc::teamDamageCounter` relies on the event either being `player_hurt` or `player_death`.
either i could make a seperate function to set the variables or completely change the flow of `Misc::teamDamageCounter` to account for `round_announce_match_start`